vql/linux/net.go

```diff
@@ -0,0 +1,101 @@
+// +build linux
+
+package linux
+
+import (
+	"context"
+	"net"
+
+	"github.com/Velocidex/ordereddict"
+
+	"www.velocidex.com/golang/velociraptor/acls"
+	"www.velocidex.com/golang/vfilter"
+	vql_subsystem "www.velocidex.com/golang/velociraptor/vql"
+
+
+
+)
+
+type LinuxNetPlugin struct {
+}
+
+// Needed values:
+// IPAddresses (with prefix)
+// MACAddress
+// On Windows, we have the following as well.  These are unimplemented on Linux ATM.
+// Subnets
+// DefaultGateway
+// DNSServer
+// DNSServerSearchOrder
+
+func (self LinuxNetPlugin) Info(scope vfilter.Scope, type_map *vfilter.TypeMap) *vfilter.PluginInfo {
+	return &vfilter.PluginInfo{
+		Name: "linux_net",
+		Doc:  "Collect network configuration on Linux clients",
+	}
+}
+
+func (self LinuxNetPlugin) Call(
+	ctx context.Context, scope vfilter.Scope,
+	args *ordereddict.Dict) <-chan vfilter.Row {
+	output_chan := make(chan vfilter.Row)
+
+	go func() {
+		defer close(output_chan)
+
+		err := vql_subsystem.CheckAccess(scope, acls.MACHINE_STATE)
+		if err != nil {
+			scope.Log("linux_net: %s", err)
+			return
+		}
+
+		interfaces, err := net.Interfaces()
```

Comment:
Is that the same as the interfaces() VQL function?

https://docs.velociraptor.app/vql_reference/windows/interfaces/

Reply:
That part is, yes. Are you suggesting that instead of using net.Interfaces() to iterate here, it should take an interface (or list of interfaces) as an argument and return the information for those? Incidentally, I suppose there's nothing inherently Linux-based about this plugin and it could just be net_info or something.

Reply:
No I was just pointing out that on Linux `SELECT * FROM inferfaces()` returns very similar data - should we just add the extra data to that plugin if there is a difference?

Reply:
Ah, I misunderstood. Yes, I can extend interfaces() to do that and drop this.